sbndqm/dqmAnalysis/PMT/SBND/CAENV1730StreamsSBND_module.cc

```diff
@@ -0,0 +1,277 @@
+////////////////////////////////////////////////////////////////////////
+//
+// 
+////////////////////////////////////////////////////////////////////////
+
+#include "art/Framework/Core/EDAnalyzer.h"
+#include "art/Framework/Core/ModuleMacros.h"
+#include "art/Framework/Principal/Event.h"
+#include "art/Framework/Principal/Handle.h"
+#include "fhiclcpp/ParameterSet.h"
+#include "canvas/Utilities/Exception.h"
+
+#include "lardataobj/RawData/OpDetWaveform.h"
+#include "larana/OpticalDetector/OpHitFinder/PMTPulseRecoBase.h"
+#include "larana/OpticalDetector/OpHitFinder/AlgoThreshold.h"
+#include "larana/OpticalDetector/OpHitFinder/AlgoSiPM.h"
+#include "larana/OpticalDetector/OpHitFinder/AlgoSlidingWindow.h"
+#include "larana/OpticalDetector/OpHitFinder/AlgoFixedWindow.h"
+#include "larana/OpticalDetector/OpHitFinder/AlgoCFD.h"
+#include "larana/OpticalDetector/OpHitFinder/PedAlgoEdges.h"
+#include "larana/OpticalDetector/OpHitFinder/PedAlgoRollingMean.h"
+#include "larana/OpticalDetector/OpHitFinder/PedAlgoUB.h"
+#include "larana/OpticalDetector/OpHitFinder/PulseRecoManager.h"
+
+#include "sbndqm/Decode/PMT/PMTDecodeData/PMTDigitizerInfo.hh"
+#include "sbndqm/Decode/Mode/Mode.hh"
+#include "sbndaq-online/helpers/SBNMetricManager.h"
+#include "sbndaq-online/helpers/MetricConfig.h"
+#include "sbndaq-online/helpers/Waveform.h"
+#include "sbndaq-online/helpers/Utilities.h"
+#include "sbndaq-online/helpers/EventMeta.h"
+
+#include <algorithm>
+#include <cassert>
+#include <stdio.h>
+#include <cmath>
+#include <fstream>
+#include <iomanip>
+#include <vector>
+#include <iostream>
+
+
+#include "messagefacility/MessageLogger/MessageLogger.h"
+
+#include "sbndqm/dqmAnalysis/TPC/FFT.hh"
+#include "sbndqm/dqmAnalysis/TPC/FFT.cc"
+
+
+namespace sbndaq {
+
+  class CAENV1730StreamsSBND : public art::EDAnalyzer {
+
+    public:
+
+        explicit CAENV1730StreamsSBND(fhicl::ParameterSet const & pset); // explicit doesn't allow for copy initialization
+  
+        virtual void analyze(art::Event const & evt);
+  
+    private:
+
+      const unsigned int nChannelsPerBoard = 16;
+      const unsigned int nBoards = 24;
+      unsigned int nTotalChannels = nBoards*nChannelsPerBoard;
+        
+        art::InputTag m_opdetwaveform_tag;
+
+        std::string m_redis_hostname;
+        
+        int m_redis_port;
+
+        double stringTime = 0.0;
+
+        fhicl::ParameterSet m_metric_config;
+
+        pmtana::PulseRecoManager pulseRecoManager;
+        pmtana::PMTPulseRecoBase* threshAlg;
+        pmtana::PMTPedestalBase*  pedAlg;
+
+        FFTManager _fft_manager;
+
+ 
+        void clean();
+
+        template<typename T> T Median( std::vector<T> data ) const ;
```

Comment:
The argument of this one should be a constant reference, and the function may be declared `static` since it does not depend on anything in the class:
```suggestion
        template<typename T> static T Median( std::vector<T> const& data );
```
That would also require changes in the implementation though.